examples/bind-shopping.amp.html

```diff
@@ -0,0 +1,454 @@
+<!doctype html>
```

Comment:
Thanks for the review!

> We need to make sure this is easy enough to validate. E.g. we can’t allow setting onclick this way.

Agree with validation. Not sure what you mean by `onclick`, did you mean the `on` attribute? Why is that a problem?

> I think we should go for double-bracing: {{}}

May I ask why? Angular uses double-bracing for string interp. while React doesn't.

> Please clarify what this means.

"Flash of unbound content" just means that binding to default attributes like `src` may cause the browser to render/load unparsed expressions. Angular describes this in its [ngBind doc](https://docs.angularjs.org/api/ng/directive/ngBind).

> A. One-time scan is not enough. We will also have to pick up deletes and other mutations via amp-live-list and other components.

> B. This scan is potentially very expensive, especially with short-cut forms such as bind-src.

Good point! Though for `amp-live-list`, we'll only have to scan the updated children rather than rescanning the entire DOM. Assuming a modest number of updates, do you think this is still a problem?

More broadly though, we certainly may need apply constraints for performance, e.g. re-evaluation frequency, cap on the absolute number of bindings, etc.

> I'd like to stress that from the model point of view, these two solutions are equivalent. Thus we first should work on reducing the algebra. I believe we can drastically reduce number of forms and expression syntax.

Agreed. :)